### PR TITLE
fix: correctly set `Dispatcher` prototype for `ProxyAgent`

### PR DESCRIPTION
**Pull request by aduh95**
In an attempt to bundle only a subset of Undici code, we forgot to take some side-effect.

Fixes: https://github.com/nodejs/corepack/issues/417
Closes: https://github.com/nodejs/corepack/pull/449

### CODE REVIEW

File: sources/httpUtils.ts [hunk 1]
```diff
@@ -100,11 +102,20 @@ async function getProxyAgent(input: string | URL) {
...
+  if (ProxyAgent == null) {
+    // Doing a deep import here since undici isn't tree-shakeable
+    const [api, Dispatcher, _ProxyAgent] = await Promise.all([
+      // @ts-expect-error internal module is untyped
+      import(`undici/lib/api/index.js`),
+      // @ts-expect-error internal module is untyped
+      import(`undici/lib/dispatcher/dispatcher.js`),
+      // @ts-expect-error internal module is untyped
+      import(`undici/lib/dispatcher/proxy-agent.js`),
+    ]);
```
Comment (arcanis) [inline sources/httpUtils.ts:114]: Are we doing all that just to reclaim some bytes? It seems a little dangerous 🤔

Comment (merceyz) [reply sources/httpUtils.ts:114]: 304000 of them but yes.

File: sources/httpUtils.ts [hunk 2]
```diff
@@ -100,11 +102,20 @@ async function getProxyAgent(input: string | URL) {
...
   if (!proxy) return undefined;
 
-  // Doing a deep import here since undici isn't tree-shakeable
-  const {default: ProxyAgent} = (await import(
-    // @ts-expect-error No types for this specific file
-    `undici/lib/proxy-agent.js`
-  )) as { default: typeof import('undici').ProxyAgent };
+  if (ProxyAgent == null) {
+    // Doing a deep import here since undici isn't tree-shakeable
+    const [api, Dispatcher, _ProxyAgent] = await Promise.all([
```
Comment (onigoetz) [inline sources/httpUtils.ts:107]: Do these need to be loaded in a particular order?

I would recommend putting them in a separate file and just exposing the ProxyAgent from that file and add a comment on why those files are needed.

Comment (aduh95) [reply sources/httpUtils.ts:107]: No, any order, that's why I'm loading them concurrently.

File: sources/httpUtils.ts (hunk 1 above)
Comment (aduh95) [reply sources/httpUtils.ts:114]: I've added tests, hopefully those ensure we'll catch bugs if this hack breaks again in the future.

**Review by arcanis — APPROVED** [review]

**Review by merceyz — APPROVED** [review]

**Pull request merged**